fastlane_core/lib/fastlane_core/build_watcher.rb

```diff
@@ -132,6 +136,20 @@ def report_status(build: nil)
           UI.message("Waiting for the build to show up in the build list - this may take a few minutes (check your email for processing issues if this continues)")
         end
       end
+
+      def force_stop_build_watching_if_required(start_time: nil, timeout_duration: nil)
+        return if start_time.nil? || timeout_duration.nil? # keep watching build for App Store Connect processing
+
+        current_time = Time.new
+        end_time = start_time + timeout_duration
+        pending_duration = end_time - current_time
+
+        if current_time > end_time
+          UI.crash!("FastlaneCore::BuildWatcher exceeded the '#{timeout_duration.to_i}' timeout duration, Stopping now!")
+        else
+          UI.message("Will 'force stop' watching build after pending #{pending_duration.to_i} timeout duration around #{end_time} time!")
```

Comment:
Thoughts on changing this to verbose? I’m not sure if printing this every time will be a bit too spammy 🤔 

```suggestion
          UI.verbose("Will 'force stop' watching build after pending #{pending_duration.to_i} timeout duration around #{end_time} time!")
```

Reply:
Yes, I agree! `verbose` looks perfect here! Thank you very much for the suggestion!